apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -108,8 +108,17 @@ type GatewayConfig struct {
 	// ControllerName is used to determine whether Contour should reconcile a
 	// GatewayClass. The string takes the form of "projectcontour.io/<namespace>/contour".
 	// If unset, the gatewayclass controller will not be started.
-	// +kubebuilder:default="projectcontour.io/projectcontour/contour"
-	ControllerName string `json:"controllerName"`
+	// Exactly one of ControllerName or GatewayName must be set.
+	// +optional
+	ControllerName string `json:"controllerName,omitempty"`
+
+	// GatewayName defines a specific Gateway that this Contour
+	// instance corresponds to. If set, Contour will reconcile
+	// only this gateway, and will not reconcile any gateway
+	// classes.
+	// Exactly one of ControllerName or GatewayName must be set.
+	// +optional
+	GatewayName *NamespacedName `json:"gatewayName,omitempty"`
```

Comment:
The naming here definitely stutters a bit, but couldn't think of a better alternative. In config this would look like:
```yaml
gateway:
  gatewayName:
    namespace: foo
    name: bar
```

One alternate option would be to call this field `specificGateway`, so:
```yaml
gateway:
  specificGateway:
    namespace: foo
    name: bar
```

Open to input!

Reply:
I like `specificGateway`, or even just `specific`. But I think people will understand if we just use `gatewayName` as well.

Reply:
I don't have a strong preference here between these options; @sunjayBhatia any thoughts?

Reply:
only thing i can think of is:

```
gateway:
  controlledGateway:
    namespace: foo
    name: bar
```

or

```
gateway:
  gatewayRef:
    namespace: foo
    name: bar
```

becomes maybe a little better if the struct name changes?

```
gatewayAPI:
  gatewayRef:
    namespace: foo
    name: bar
```

not a too strong opinion, your other options are good too, favorite of mine is maybe `gatewayRef`

Reply:
I am +1 on `gatewayRef` - `xRef` is the preferred upstream way to name this field, so it's consistent with upstream.

Reply:
:+1: I like that, good idea @sunjayBhatia! Have updated.